java/arcs/core/util/Scheduler.kt

```diff
@@ -285,6 +290,16 @@ class Scheduler(
         ) : Scheduler.Task.Listener("dispatcher", "non-particle", block)
     }
 
+    private val currentDispatcherThreadLocal = ThreadLocal<CoroutineDispatcher?>()
```

Comment:
Fields go at the top of the class

Reply:
Also, maybe instead of being a reference to the dispatcher itself, it could be just the dispatcher's hashcode? Less opportunity to leak a dispatcher somehow that way.